tortoise/backends/aioodbc/client.py

```diff
@@ -0,0 +1,272 @@
+import asyncio
+from functools import wraps
+from typing import Any, Callable, List, Optional, SupportsInt, Tuple, TypeVar, Union
+
+import aioodbc
+import pyodbc
+from pypika import Dialects
+from pypika.queries import Query, QueryBuilder
+
+from tortoise.backends.aioodbc.executor import AioodbcExecutor
+from tortoise.backends.aioodbc.schema_generator import AioodbcSchemaGenerator
+from tortoise.backends.base.client import (
+    BaseDBAsyncClient,
+    BaseTransactionWrapper,
+    Capabilities,
+    ConnectionWrapper,
+    NestedTransactionPooledContext,
+    PoolConnectionWrapper,
+    TransactionContext,
+    TransactionContextPooled,
+)
+from tortoise.exceptions import (
+    DBConnectionError,
+    IntegrityError,
+    OperationalError,
+    TransactionManagementError,
+)
+
+FuncType = Callable[..., Any]
+F = TypeVar("F", bound=FuncType)
+
+
+def translate_exceptions(func: F) -> F:
+    @wraps(func)
+    async def translate_exceptions_(self, *args):
+        try:
+            return await func(self, *args)
+        except pyodbc.ProgrammingError as exc:
+            raise OperationalError(exc)
```

Comment:
I wish pyodbc differentiated between 
`24***`, `42***` & `25***` as the `25***` would map very well to our `TransactionManagementError`

It seems pyodbc stores the `SQLSTATE` in the exception: https://github.com/mkleehammer/pyodbc/blob/a3693bf175a32e6c3cea9c5c3be65acdae5467f2/src/errors.cpp#L88

Is it possible to inspect it, and if `25***` raise `TransactionManagementError` instead?

Reply:
Definitely, I will look into that. I was fighting this same issue to catch timeouts in pools and ended up checking for keywords in the error message as oracle wasn't returning a sqlstate worth anything AFAIRemember.